schema/query.go

```diff
@@ -354,25 +358,33 @@ func (e NotExist) Match(payload map[string]interface{}) bool {
 // Match implements Expression interface
 func (e GreaterThan) Match(payload map[string]interface{}) bool {
 	n, ok := isNumber(getField(payload, e.Field))
-	return ok && (n > e.Value)
+	if ok {
+		return ok && (n > e.Value.(float64))
+	}
+	t, ok := isTime(getField(payload, e.Field))
+	if ok {
```

Comment:
The `e.Value` should be casted to `time.Time` by `validateQuery` and `isTime` does not seem to check if the passed value is already a `time.Time`.

Also, as the data in the payload should be stored as casted by the validator (at least with mem storage which is the only storer to use those methods), shouldn't it be stored as a `time.Time` too?

Looks like the only case were those condition would work is when we compare a non time field with a time-like value.

In my understanding the condition should be written like this:

```go
func (e GreaterThan) Match(payload map[string]interface{}) bool {
	val := getField(payload, e.Field)
	if val, ok := isNumber(val); ok {
		if cond, ok := isNumber(e.Value) {
			return val > cond
		}
	}
	if val, ok := val.(time.Time); ok {
		if cond, ok := e.Value.(time.Time); ok {
			return val.After(cond)
		}
	}
	return false
}
```